services/api/src/resources/deployment/resolvers.ts

```diff
@@ -143,6 +143,59 @@ export const getDeploymentsByBulkId: ResolverFn = async (
   return withK8s;
 };
 
+export const getDeploymentsByFilter: ResolverFn = async (
+  root,
+  input,
+  { sqlClientPool, hasPermission, models, keycloakGrant }
+) => {
+
+  const { openshifts, deploymentStatus = ["NEW", "PENDING", "RUNNING"] } = input;
+
+  /*
+    use the same mechanism for viewing all projects
+    bulk deployments can span multiple projects, and anyone with access to a project
+    will have the same rbac as `deployment:view` which covers all possible roles for a user
+    otherwise it will only return all the projects they have access to
+    and the listed deployments are sourced accordingly to which project
+    the user has access to
+  */
+  let userProjectIds: number[];
+  try {
+    await hasPermission('project', 'viewAll');
+  } catch (err) {
+    if (!keycloakGrant) {
+      logger.warn('No grant available for getAllProjects');
+      return [];
+    }
+
+    userProjectIds = await models.UserModel.getAllProjectsIdsForUser({
+      id: keycloakGrant.access_token.content.sub
+    });
+  }
+
+  let queryBuilder = knex.select("deployment.*").from('deployment').
+      join('environment', 'deployment.environment', '=', 'environment.id').
+      join('project', 'environment.project', '=', 'project.id');
+
+  if(openshifts) {
+    queryBuilder = queryBuilder.whereIn('deployment.remote_id', openshifts);
```

Comment:
Should this be `whereIn('environment.openshift', openshifts)` ?

The `remote_id` is just a uuid that is used for tracking logs mainly.

Reply:
Well, this query works :D Resolved.